packages/firestore/src/lite/reference.ts

```diff
@@ -48,16 +49,96 @@ export interface DocumentData {
   [field: string]: any;
 }
 
+/** Primitive types. */
+export type Primitive = string | number | boolean | undefined | null;
+
+/**
+ * Similar to Typescript's `Partial<T>`, but allows nested fields to be
+ * omitted and FieldValues to be passed in as property values.
+ */
+export type NestedPartial<T> = T extends Primitive
+  ? T
+  : T extends Map<infer K, infer V>
+  ? Map<NestedPartial<K>, NestedPartial<V>>
+  : T extends {}
+  ? { [K in keyof T]?: NestedPartial<T[K]> | FieldValue }
+  : Partial<T>;
+
+/**
+ * Allows FieldValues to be passed in as a property value while maintaining
+ * type safety.
+ */
+export type WithFieldValue<T> = T extends Primitive
+  ? T
+  : T extends {}
+  ? { [K in keyof T]: WithFieldValue<T[K]> | FieldValue }
+  : Partial<T>;
+
+/**
+ * Update data (for use with {@link @firebase/firestore/lite#(updateDoc:1)})
```

Comment:
We use `{@link (updateDoc:1)}` in other places, which does not hardcode the lite docs. Would that work here?